apps/site/assets/ts/hooks/usePageVisibility.ts

```diff
@@ -0,0 +1,19 @@
+/* eslint-disable import/prefer-default-export */
+import { useState, useLayoutEffect } from "react";
+
+/** returns true if the page is visible
+ * returns false if the page is hidden
+ */
+export const usePageVisibility = (): boolean => {
+  const [hidden, setHidden] = useState<boolean>(false);
+  const callback = (): void => {
+    setHidden(document.hidden);
+  };
+  useLayoutEffect(() => {
+    document.addEventListener("visibilitychange", callback);
+    return () => {
+      document.removeEventListener("visibilitychange", callback);
```

Comment:
question: Don't we always want to be listening for visibility changes to know wether to leave or join channels?

Reply:
yeah, probably!